app/src/main/java/org/wikipedia/dataclient/okhttp/OfflineCacheInterceptor.java

```diff
@@ -50,6 +61,22 @@ public Response intercept(@NonNull Chain chain) throws IOException {
             cacheCandidate = cacheDelegate.internalCache().get(request);
         }
 
+        if (cacheCandidate == null  && request.url().toString().contains(RestService.PAGE_HTML_ENDPOINT)) {
+            int indexOfTitleDelimiter = request.url().toString().lastIndexOf('/');
+            File file = new File(WikipediaApp.getInstance().getFilesDir() + "/" + CONVERTED_FILES_DIRECTORY_NAME + "/" + request.url().toString().substring(indexOfTitleDelimiter + 1));
+            if (file.exists()) {
+                FileInputStream fileInputStream = new FileInputStream(file);
+                String contentJSON = FileUtil.readFile(fileInputStream);
+                cacheCandidate = new Response.Builder()
+                        .code(STATUS_CODE_SUCCESS)
+                        .request(request)
+                        .protocol(HTTP_1_1)
+                        .message("")
+                        .body(ResponseBody.create(MEDIA_JSON, contentJSON))
```

Comment:
Hint: you're saying that the mime-type of this response is JSON. But... is it actually JSON?

Reply:
yes! i was changing the 'content-type' header but i guess that was getting overwritten... thank you 😅